### PR TITLE
Optimize tmva includes

### PR DESCRIPTION
**Pull request by linev**
Use RtypesCore.h where Rtypes.h not required
Place `<vector>`, `<string>`, `<map>`, `<utility>` includes where
correspondent standard classes are used
In rare cases remove such unused standard includes
Remove `TString.h` where not used
Also remove three temporary files which accidentally comes to repository

### CODE REVIEW

File: tmva/tmva/inc/TMVA/Envelope.h [hunk 1]
```diff
@@ -12,8 +12,9 @@
 #define ROOT_TMVA_Envelope
 
 #include <sstream>
-#include<iostream>
+#include <iostream>
```
Comment (stwunsch) [inline tmva/tmva/inc/TMVA/Envelope.h:15]: I don't see why this is required.

File: tmva/tmva/inc/TMVA/OptionMap.h [hunk 2]
```diff
@@ -12,10 +12,9 @@
 #define ROOT_TMVA_OptionMap
 
 #include <sstream>
-#include<iostream>
-#include<map>
-
-#include <TNamed.h>
+#include <iostream>
```
Comment (stwunsch) [inline tmva/tmva/inc/TMVA/OptionMap.h:15]: Also here, if i see correctly `iostream` is not required.

File: tmva/tmva/inc/TMVA/ROCCurve.h [hunk 3]
```diff
@@ -23,12 +23,13 @@
 #ifndef ROOT_TMVA_ROCCurve
 #define ROOT_TMVA_ROCCurve
 
-#include "Rtypes.h"
+#include "RtypesCore.h"
 
 #include <iomanip>
 #include <iostream>
```
Comment (stwunsch) [inline tmva/tmva/inc/TMVA/ROCCurve.h:29]: And here also an unused `iostream`

File: tmva/tmva/inc/TMVA/Envelope.h (hunk 1 above)
Comment (linev) [reply tmva/tmva/inc/TMVA/Envelope.h:15]: Yes, you are right!
I forgot to check iostream in header files - only in sources :(

**Comment by linev** [comment]
Failures have nothing to do with commit - some strange ACLiC problems

**Pull request merged**